pkg/controller/operators/catalog/operator.go

```diff
@@ -487,36 +506,59 @@ func (o *Operator) syncDependentSubscriptions(logger *logrus.Entry, catalogSourc
 	}
 }
 
-func (o *Operator) syncSubscriptions(obj interface{}) error {
-	sub, ok := obj.(*v1alpha1.Subscription)
+func (o *Operator) syncResolvingNamespace(obj interface{}) error {
+	ns, ok := obj.(*corev1.Namespace)
 	if !ok {
 		o.Log.Debugf("wrong type: %#v", obj)
-		return fmt.Errorf("casting Subscription failed")
+		return fmt.Errorf("casting Namespace failed")
 	}
-	namespace := sub.GetNamespace()
+	namespace := ns.GetName()
 
 	logger := o.Log.WithFields(logrus.Fields{
-		"sub":       sub.GetName(),
-		"namespace": sub.GetNamespace(),
-		"source":    sub.Spec.CatalogSource,
-		"pkg":       sub.Spec.Package,
-		"channel":   sub.Spec.Channel,
+		"namespace": namespace,
+		"id":        queueinformer.NewLoopID(),
 	})
 
-	// record the current state of the desired corresponding CSV in the status. no-op if we don't know the csv yet.
-	sub, err := o.ensureSubscriptionCSVState(logger, sub)
+	// get the set of sources that should be used for resolution and best-effort get their connections working
+	logger.Debug("resolving sources")
+	resolverSources := o.ensureResolverSources(logger, namespace)
+
+	logger.Debug("checking if subscriptions need update")
+
+	subs, err := o.lister.OperatorsV1alpha1().SubscriptionLister().Subscriptions(namespace).List(labels.Everything())
 	if err != nil {
+		logger.WithError(err).Debug("couldn't list subscriptions")
 		return err
 	}
 
-	// return early if the subscription is up to date
-	if o.nothingToUpdate(logger, sub) {
+	shouldUpdate := false
+	for _, sub := range subs {
+		logger := logger.WithFields(logrus.Fields{
+			"sub":     sub.GetName(),
+			"source":  sub.Spec.CatalogSource,
+			"pkg":     sub.Spec.Package,
+			"channel": sub.Spec.Channel,
+		})
+
+		// ensure the installplan reference is correct
+		sub, err := o.ensureSubscriptionInstallPlanState(logger, sub)
+		if err != nil {
+			return err
+		}
+
+		// record the current state of the desired corresponding CSV in the status. no-op if we don't know the csv yet.
+		sub, err = o.ensureSubscriptionCSVState(logger, sub)
+		if err != nil {
+			return err
+		}
+		shouldUpdate = shouldUpdate || !o.nothingToUpdate(logger, sub)
```

Comment:
nit: I think this can just be `!o.nothingToUpdate(...)`

Reply:
if we do `shouldUpdate = !nothingToUpdate`, then a later iteration can "reset" `shouldUpdate` to `false`.

Reply:
Ah, I see - missed that this was in a loop. Looks good.